src/bp/ui-studio/src/web/reducers/flows.ts

```diff
@@ -214,12 +232,70 @@ const doCreateNewFlow = name => ({
   ]
 })
 
+const handleServerRename = (state, modification) => {
+  if (!modification.newName || _.keys(state.flowsByName).includes(modification.newName)) {
+    return
+  }
+  return modification
+}
+
+const handleServerCreate = (state, modification) => {
+  if (_.keys(state.flowsByName).includes(modification.name)) {
+    return
+  }
+  return modification
+}
+
+const handleServerDelete = (state, modification) => {
+  if (!_.keys(state.flowsByName).includes(modification.name)) {
+    return
+  }
+  return modification
+}
+
+const handleServerUpdate = (state, modification) => {
+  const flowHash = computeHashForFlow(modification.payload)
+  const currentFlowHash = computeHashForFlow(state.flowsByName[modification.name])
+  if (currentFlowHash === flowHash) {
+    return
+  }
+
+  return modification
+}
+
 // *****
 // Reducer that deals with non-recordable (no snapshot taking)
 // *****
 
 let reducer = handleActions(
   {
+    [receiveFlowsModification]: (state, { payload }) => {
+      let actualModification
+      switch (payload.modification) {
+        case 'create':
+          actualModification = handleServerCreate(state, payload)
+          break
+        case 'delete':
+          actualModification = handleServerDelete(state, payload)
+          break
+        case 'update':
+          actualModification = handleServerUpdate(state, payload)
+          break
+        case 'rename':
+          actualModification = handleServerRename(state, payload)
```

Comment:
I'd keep all your handle* logic inside this reducer, it doesn't handle anything, it just returns the modification that was done.

Reply:
Well, I don't mind keeping all the logic inside this reducer, but it'll be a big chunk of hard to read code.